portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,68 @@
 
 package com.google.sps.servlets;
 
+import com.google.sps.servlets.Comment;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.List;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comments data. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<Comment> comments = new ArrayList<Comment>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+
+    // Convert the comments into JSON.
+    String json = convertToJson(comments);
+
+    // Send the JSON as the response.
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get input from the form.
+    String text = getParameter(request, "comment-text", "");
+    String commenterName = getParameter(request, "commenter-name", "Anonymous");
+    String commenterEmail = getParameter(request, "commenter-email", "Unknown");
+
+    // Create new Comment object to store input.
+    Comment newComment = new Comment(text, commenterName, commenterEmail);
+
+    // Add comment to the comments data.
+    comments.add(newComment);
+
+    // Redirect to same HTML page.
+    response.sendRedirect("/index.html");
+  }
+
+  /*
+   * Converts List of comments into a JSON using the gson library.
+   */
+  private String convertToJson(List<Comment> comments){
+    Gson gson = new Gson();
+    return gson.toJson(comments);
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
```

Comment:
The format of comment is very strange. Did you use any auto formatter?

Reply:
Done -  removed the extra asterisk at the top. Yes, prettier did not flag this. I tried to follow the example here since this is javadoc. https://google.github.io/styleguide/javaguide.html#s7-javadoc

Reply:
Ah I see.